test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/SignCommandTestFixture.cs

```diff
@@ -28,13 +32,53 @@ public TrustedTestCert<TestCertificate> TrustedTestCertificate
             {
                 if (_trustedTestCert == null)
                 {
-                    _trustedTestCert = TestCertificate.Generate().WithPrivateKeyAndTrust();
+                    Action<X509V3CertificateGenerator> actionGenerator = delegate (X509V3CertificateGenerator gen)
+                    {
+                        // CodeSigning EKU
+                        var usages = new[] { KeyPurposeID.IdKPCodeSigning };
+
+                        gen.AddExtension(
+                            X509Extensions.ExtendedKeyUsage.Id,
+                            critical: true,
+                            extensionValue: new ExtendedKeyUsage(usages));
+                    };
+
+                    // Code Sign EKU needs trust to a root authority
+                    // Add the cert to Root CA list in LocalMachine as it does not prompt a dialog
+                    // This makes all the associated tests to require admin privilege
+                    _trustedTestCert = TestCertificate.Generate(actionGenerator).WithPrivateKeyAndTrust(StoreName.Root, StoreLocation.LocalMachine);
                 }
 
                 return _trustedTestCert;
             }
         }
 
+        public TrustedTestCert<TestCertificate> TrustedTestCertificateWithInvalidEku
+        {
+            get
+            {
+                if (_trustedTestCertWithInvalidEku == null)
+                {
+                    Action<X509V3CertificateGenerator> actionGenerator = delegate (X509V3CertificateGenerator gen)
```

Comment:
I see this same code duplicated for packaging tests, which isn't awful for tests. But if you think this will be used in more places I suggest adding a common utility for this to test.utility.